Utilities/build-script-helper.py

```diff
@@ -8,230 +8,306 @@
+    env['SOURCEKIT_TOOLCHAIN_PATH'] = args.toolchain
+    env['INDEXSTOREDB_TOOLCHAIN_BIN_PATH'] = args.toolchain
+    # Use local dependencies (i.e. checked out next sourcekit-lsp).
+    if not args.no_local_deps:
+        env['SWIFTCI_USE_LOCAL_DEPS'] = "1"
+
+    if args.ninja_bin:
+        env['NINJA_BIN'] = args.ninja_bin
+
+    if args.sanitize and 'address' in args.sanitize:
+        # Workaround reports in Foundation: https://bugs.swift.org/browse/SR-12551
+        env['ASAN_OPTIONS'] = 'detect_leaks=false'
+    if args.sanitize and 'undefined' in args.sanitize:
+        supp = os.path.join(args.package_path, 'Utilities', 'ubsan_supressions.supp')
+        env['UBSAN_OPTIONS'] = 'halt_on_error=true,suppressions=%s' % supp
+    if args.sanitize and 'thread' in args.sanitize:
+        env['TSAN_OPTIONS'] = 'halt_on_error=true'
+
+    if args.action == 'test' and not args.skip_long_tests:
+        env['SOURCEKIT_LSP_ENABLE_LONG_TESTS'] = '1'
+
+    env['SWIFT_EXEC'] = '%sc' % (swift_exec)
+
+    return env
+
+
+def build(swift_exec: str, args: argparse.Namespace) -> None:
+    """
+    Build one product in the package
+    """
+    swiftpm_args = get_swiftpm_options(swift_exec, args)
+    env = get_swiftpm_environment_variables(swift_exec, args)
+    cmd = [swift_exec, 'build'] + swiftpm_args
+    check_call(cmd, env=env, verbose=args.verbose)
+
+
+def run_tests(swift_exec: str, args: argparse.Namespace) -> None:
+    """
+    Run all tests in the package
+    """
+    swiftpm_args = get_swiftpm_options(swift_exec, args)
+    env = get_swiftpm_environment_variables(swift_exec, args)
+
     bin_path = swiftpm_bin_path(swift_exec, swiftpm_args, env)
     tests = os.path.join(bin_path, 'sk-tests')
     print('Cleaning ' + tests)
     shutil.rmtree(tests, ignore_errors=True)
-    test_args = swiftpm_args
-    test_args += ['--parallel', '--disable-testable-imports']
-    swiftpm('test', swift_exec, test_args, env)
-  elif args.action == 'install':
+    
+    cmd = [
+        swift_exec, 'test', 
+        '--parallel', 
+        '--disable-testable-imports',
+        '--test-product', 'SourceKitLSPTests'
+    ] + swiftpm_args
+    check_call(cmd, env=env, verbose=args.verbose)
+
+
+def swiftpm(action: str, swift_exec: str, swiftpm_args: List[str], env: Optional[Dict[str, str]]=None) -> None:
+    cmd = [swift_exec, action] + swiftpm_args
+    print(' '.join(cmd))
+    subprocess.check_call(cmd, env=env)
+
+def install_binary(exe: str, source_dir: str, install_dir: str, verbose: bool) -> None:
+    cmd = ['rsync', '-a', os.path.join(source_dir, exe), install_dir]
+    check_call(cmd, verbose=verbose)
+
+def install(swift_exec: str, args: argparse.Namespace) -> None:
+    swiftpm_args = get_swiftpm_options(swift_exec, args)
+    env = get_swiftpm_environment_variables(swift_exec, args)
+
     bin_path = swiftpm_bin_path(swift_exec, swiftpm_args, env)
     swiftpm_args += ['-Xswiftc', '-no-toolchain-stdlib-rpath']
```

Comment:
Incidentally, I guess this causes rebuilds during `install` actions since the arguments have changed?

Reply:
Yes, I think so. Looking into the actual build time of SourceKit-LSP is the next thing I want to do.